test/export/test_export.py

```diff
@@ -3552,6 +3552,18 @@ def forward(self, x):
             )
         )
 
+    def test_use_embedding_twice(self):
```

Comment:
Hi Pian, not directly related to this Diff, but I'm wondering, if I run `torch.export._trace._export_for_training` on this graph, do I expect the following result, where embed_weight_1 is used twice?

```
graph():
    %embed_weight : [num_users=0] = get_attr[target=embed.weight]
    %embed_weight_1 : [num_users=2] = get_attr[target=embed.weight]
    %x : [num_users=2] = placeholder[target=x]
    %embedding : [num_users=1] = call_function[target=torch.ops.aten.embedding.default](args = (%embed_weight_1, %x), kwargs = {})
    %index : [num_users=1] = call_function[target=torch.ops.aten.index.Tensor](args = (%embed_weight_1, [%x]), kwargs = {})
    %add : [num_users=1] = call_function[target=torch.ops.aten.add.Tensor](args = (%embedding, %index), kwargs = {})
    return (add,)
```

If I run the default `capture_pre_autigrad_graph`, I get the following, where there's no parameter re-use.

```
graph():
    %arg0_1 : [num_users=2] = placeholder[target=arg0]
    %_param_constant0 : [num_users=1] = get_attr[target=embed_weight]
    %embedding : [num_users=1] = call_function[target=torch.ops.aten.embedding.default](args = (%_param_constant0, %arg0_1), kwargs = {})
    %_param_constant0_1 : [num_users=1] = get_attr[target=embed_weight]
    %index : [num_users=1] = call_function[target=torch.ops.aten.index.Tensor](args = (%_param_constant0_1, [%arg0_1]), kwargs = {})
    %add : [num_users=1] = call_function[target=torch.ops.aten.add.Tensor](args = (%embedding, %index), kwargs = {})
    return [add]
```

Asking because this discrepancy caused some downstream error in training ir migration. Thanks!

cc @tugsbayasgalan

Reply:
btw I just want to confirm if this is the expected behavior. If it is, we can try to fix the downstream code.

Reply:
(resolved in chat, the training ir behavior is correct.)